Functions/xASL_Iteration.m

```diff
@@ -306,31 +330,34 @@
         
         % First check if this iteration has been fully processed, then we will skip the logging for this iteration
         if  exist(xASL_adm_ReplaceSymbols(fullfile(x.dir.LockDir,x.settings.MUTEXID,'999_ready.status'),x),'file')
-            AlreadyProcessed    = 1;
+            AlreadyProcessed = 1;
         else
-            AlreadyProcessed    = 0;
+            AlreadyProcessed = 0;
         end
         
         
-        % optional diary logfile of command window
-
+        % Optional diary logfile of command window
         if ~isempty(diaryFile)
+            % NB. Don't overwrite diaryFile itself because it might be expanded
+            % with different values in the next iteration.
             diaryFileEx = xASL_adm_ReplaceSymbols(diaryFile,x);
-                % NB. Don't overwrite diaryFile itself because it might be expanded
-                % with different values in the next iteration.
-            status = xASL_adm_CreateDir(fileparts(diaryFileEx),2); % make sure folder exists, but limit to 2 missing levels
+            diaryFileEx = fullfile(diaryFileEx);
```

Comment:
Also no idea what is happening...

Reply:
Oh, eeeehmm 😄 

So I did this right [here](https://github.com/ExploreASL/ExploreASL/commit/552aea29f8544ac1f136f80cb1e7f725adec081b#diff-dbe9affae9c7d0938ce3c68c6890f26d105b43cc9dc4239b4c16e16e91041d57) just to simplify the code a bit. `fullfile` is used to fix the windows/unix related paths. And `fileparts` is used to get the directory.

**Windows**
```matlab
>> test_path = 'C:\users\michael\i\am\a\test\logfile.log';
>> test_path_ex = fullfile(test_path_ex)
test_path_ex =
    'C:\users\michael\i\am\a\test\logfile.log'

```

**Unix**
```matlab
>> test_path = 'C:/users/michael/i/am/a/test/logfile.log';
>> test_path_ex = fullfile(test_path_ex)
test_path_ex =
    'C:\users\michael\i\am\a\test\logfile.log'
```

So before we did ...

```
status = xASL_adm_CreateDir(fileparts(diaryFileEx),2);
```

... and now it does ...

```
diaryFileEx = fullfile(diaryFileEx);
status = xASL_adm_CreateDir(fullfile(fileparts(diaryFileEx)),2);
```

... but I agree that this could have been done in a better way and not using `fullfile` twice.
Maybe...

```
diaryFileEx = fullfile(diaryFileEx);
status = xASL_adm_CreateDir(fileparts(diaryFileEx),2);
```

... ? I just found out that it sometimes printed paths that were 50% windows and 50% unix. Still something unclear?

Reply:
Fix 46feecd  ?

Reply:
Ah OK I didn't know it would print paths that had incorrect file separators. Nice!